### DISCUSSION TITLE
Update OneDrive Running like Service

### DISCUSSION THREAD

**Discussion by marcoseguracr** [abraunegg/onedrive#2116] [Q&A]
Hi,

I tested your tool and is amazing, I got it working without issues quick on Amazon Linux v2 (compile) and get sync easier. I cloned the machine to create a new service that use new configuration (driveID and new username). I am running the onedrive like service and this is my case:

On cloned server
1- Stopped onedrive service
2- logout onedrive
3- clear the sync folder from previous server
4- execute "onedrive" from command to get logged again with new username (I user incognito to warranty I am logged with new sync username)
5- Updated the config file to match the new driveID
6- make a --display-config to validate my configuration and all was good
7- execute a synchronization with --dry-run for testing (all good here)
8- execute synchronize with --resync to validate all works without issues (I got a test file saved into my new sharepoint library with new username)
9- Started the onedrive service and here all fails, because looks like the service have saved the configuration (login token) somewhere.

I remember on redhat linux forks (like Amazon Linux) that there is a special command to run command like a service, in that way the token can be logout and login back with new user). I tested my configuration on service and I am 100% sure is using my ~/.config/onedrive/config, but look like services run different.

I told write my test steps, but in my original setup I faced same issue, but in one of the onedrive channels on reddit somebody give me the right solution and I executed command like service and I was able to switch from my username (I was using that on the beginning) to the right one, without rebuild the machine.

I really appreciate if you face this issue when running like a service. In the meantime I patched by executing a crontab that runs every minute (like the service) and works without issues, but that is not the right way.

Thank you

Marco

**Comment by abraunegg** [comment]
@marcoseguracr 
To run the client as a service, on almost any more modern platforms - Amazon Linux included - you need to configure the systemd service to do so. Please read: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-as-a-system-service

When you are running the client, do not use 'sudo' ... as this elevates your client to 'root' - thus, it looks for the 'config directory' in 'root' - rather than your home directory.

Additionally, if you are running via crontab - please dont do that - it is also running the client as 'root' .. rather than your normal user.

You also seem to be trying to configure multiple client configurations + SharePoint. Please read:
* https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-to-use-multiple-onedrive-accounts--configurations
* https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md

**Comment by bellessecretsshow** [comment]
> Hi,
> 
> I tested your tool and is amazing, I got it working without issues quick on Amazon Linux v2 (compile) and get sync easier. I cloned the machine to create a new service that use new configuration (driveID and new username). I am running the onedrive like service and this is my case:
> 
> On cloned server 1- Stopped onedrive service 2- logout onedrive 3- clear the sync folder from previous server 4- execute "onedrive" from command to get logged again with new username (I user incognito to warranty I am logged with new sync username) 5- Updated the config file to match the new driveID 6- make a --display-config to validate my configuration and all was good 7- execute a synchronization with --dry-run for testing (all good here) 8- execute synchronize with --resync to validate all works without issues (I got a test file saved into my new sharepoint library with new username) 9- Started the onedrive service and here all fails, because looks like the service have saved the configuration (login token) somewhere.
> 
> I remember on redhat linux forks (like Amazon Linux) that there is a special command to run command like a service, in that way the token can be logout and login back with new user). I tested my configuration on service and I am 100% sure is using my ~/.config/onedrive/config, but look like services run different.
> 
> I told write my test steps, but in my original setup I faced same issue, but in one of the onedrive channels on reddit somebody give me the right solution and I executed command like service and I was able to switch from my username (I was using that on the beginning) to the right one, without rebuild the machine.
> 
> I really appreciate if you face this issue when running like a service. In the meantime I patched by executing a crontab that runs every minute (like the service) and works without issues, but that is not the right way.
> 
> Thank you
> 
> Marco

i managed to get it working  in an lxc    for 1 account    

https://github.com/abraunegg/onedrive/discussions/2144
